example/lib/main-search.dart

```diff
@@ -0,0 +1,344 @@
+                        child: IconButton(
+                          onPressed: () {
+                            setState(() {
+                              _emojiShowing = !_emojiShowing;
+                            });
+                          },
+                          icon: const Icon(
+                            Icons.emoji_emotions,
+                            color: Colors.white,
+                          ),
+                        ),
+                      ),
+                      Expanded(
+                        child: Padding(
+                          padding: const EdgeInsets.symmetric(vertical: 8.0),
+                          child: TextField(
+                              controller: _controller,
+                              style: const TextStyle(
+                                  fontSize: 20.0, color: Colors.black87),
+                              decoration: InputDecoration(
+                                hintText: 'Type a message',
+                                filled: true,
+                                fillColor: Colors.white,
+                                contentPadding: const EdgeInsets.only(
+                                    left: 16.0,
+                                    bottom: 8.0,
+                                    top: 8.0,
+                                    right: 16.0),
+                                border: OutlineInputBorder(
+                                  borderRadius: BorderRadius.circular(50.0),
+                                ),
+                              )),
+                        ),
+                      ),
+                      Material(
+                        color: Colors.transparent,
+                        child: IconButton(
+                            onPressed: () {
+                              // send message
+                            },
+                            icon: const Icon(
+                              Icons.send,
+                              color: Colors.white,
+                            )),
+                      )
+                    ],
+                  )),
+              Offstage(
+                offstage: !_emojiShowing,
+                child: ValueListenableBuilder<TextEditingValue>(
+                    valueListenable: _searchController,
+                    builder: (context, value, _) {
+                      return Column(
+                        children: [
+                          if (value.text.isEmpty)
+                            SizedBox(
+                                height: 250,
+                                child: EmojiPicker(
+                                  textEditingController: _controller,
+                                  config: _config,
+                                ))
+                          else
+                            _buildSearchResults(context, emojiSize, cellSize),
+                          _buildSearchBar(context, value.text.isEmpty)
```

Comment:
Currently it's not possible to remove the focus from the searchBar. How about aligning it with Whatsapp keyboard that collapse to SearchResults after pressing on the search-icon already and provides a back button to show the emoji-keyboard again. Just an suggestion since you were not happy with the current example.
![Screenshot_20220825-222627](https://user-images.githubusercontent.com/11667234/186678366-cf7cfdd3-ba89-436d-a7a8-f50b7332f9d4.png)

Reply:
I've added a back button. It's not exactly as in Whatsapp: they, for example show some common emojis with empty search text. I think it's a good start, but there are so many usability options you can think about which in the end come down to personal preferences. That's why search implementation outside of the library itself makes a lot of sense.